internal/controller/impersonatorconfig/impersonator_config.go

```diff
@@ -521,37 +524,109 @@ func (c *impersonatorConfigController) ensureClusterIPServiceIsStopped(ctx conte
 	return utilerrors.FilterOut(err, k8serrors.IsNotFound)
 }
 
-func (c *impersonatorConfigController) createOrUpdateService(ctx context.Context, service *v1.Service) error {
-	log := c.infoLog.WithValues("serviceType", service.Spec.Type, "service", klog.KObj(service))
-	existing, err := c.servicesInformer.Lister().Services(c.namespace).Get(service.Name)
+func (c *impersonatorConfigController) createOrUpdateService(ctx context.Context, desiredService *v1.Service) error {
+	log := c.infoLog.WithValues("serviceType", desiredService.Spec.Type, "service", klog.KObj(desiredService))
+
+	// Prepare to remember which annotation keys were added from the CredentialIssuer spec, both for
+	// creates and for updates, in case someone removes a key from the spec in the future. We would like
+	// to be able to detect that the missing key means that we should remove the key. This is needed to
+	// differentiate it from a key that was added by another actor, which we should not remove.
+	// But don't bother recording the requested annotations if there were no annotations requested.
+	desiredAnnotationKeys := []string{}
```

Comment:
Minor nit but you can do `make([]string, 0, len(desiredService.Annotations))` to allocate this slice memory all at once.

This may get flagged by the [`prealloc`](https://github.com/alexkohler/prealloc#purpose) lint checker.

Reply:
Fixed in f17f7c0c6a53f50e6933a796b0804eb119a671cf.